### PR TITLE
fix: don't override GITHUB_TOKEN

### PR DESCRIPTION
**Pull request by jsdalton**
## Purpose

* This action is not working in repository's that use the Github [Merge Queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) feature. Example failure: https://github.com/contentful/marketplace-partner-apps/actions/runs/7176568700/job/19541648881?pr=572
* The error message is "GraphQL: Changes must be made through the merge queue (mergePullRequest)". According to this [Github Issue](https://github.com/cli/cli/issues/8352) the error stems from a platform issue that occurs when using the GIthub CLI with a personal access token

## Approach

* Remove the override of the `GITHUB_TOKEN` value for this merge job. Github already provides a `GITHUB_TOKEN` to the environment (see https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token), which can be managed via worker permissions instead of on the token itself.

### CODE REVIEW

**Comment by jamesmbourne** [comment]
@contentful/team-mechagodzilla anyone recall why we override the GH Token here? I believe we considered using the built-in GH Token but chose not to for a reason which I don't remember 😬

**Comment by jamesmbourne** [comment]
@jsdalton it's a bit trick to test this kind of thing automatically, have you validated this works by referencing the commit SHA/branch of the action from another repo?

In general, we prefer to use the built-in GH Actions token where possible, so the fact we fetch one from Vault suggests there may have been some limitation of the built-in token we needed to work around, but I'm not sure if this was the case.

**Comment by jsdalton** [comment]
I was wondering the same @jamesmbourne . It look like we've had it there since this file's inception: https://github.com/contentful/github-auto-merge/blob/41cff919810b975ad7689904fc2fba13fabd2e9f/action.yml#L60-L61

I wonder if either Github didn't at that time provide `GITHUB_TOKEN` to actions automatically as they do today? Or maybe we just didn't realize at the time that it wasn't necessary.

The big advantage of using the provided token is you have the ability to manage permissions directly, i.e. via https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

**Comment by jamesmbourne** [comment]
Happy to give it a try @jsdalton 👍 

Looks like this line will need to be updated to check for comments from `github-actions[bot]`
https://github.com/contentful/github-auto-merge/blob/main/action.yml#L36-L39

There was a previous version of this action https://github.com/contentful/dependabot-auto-merge but that doesn't provide many clues about why we didn't use the built-in token either.

**Comment by jsdalton** [comment]
@jamesmbourne Looking a bit closer, I wonder if we shouldn't with this PR provide the explicit permissions needed to the job?

Right now we have been doing this in the project where we bring in the action, but I'm thinking it makes more sense to do it with the job itself (i.e. so the job declares the permissions it needs): https://github.com/contentful/marketplace-partner-apps/blob/main/.github/workflows/dependabot-approve-and-request-merge.yml#L6-L10. WDYT?

**Comment by jamesmbourne** [comment]
@jsdalton yes I think we should specify the permissions - the defaults can be found here: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

**Comment by jsdalton** [comment]
@jamesmbourne  Hmm in theory it sounded correct but in practice I guess GH actions don't work like that 🤷 

There's no place in the metadata syntax guide where we can specify permissions: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs

Instead I assume you're supposed to do it the way we did it in the link I shared. When you specify a shared action as a job you provide the permissions there: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

But the good news is we already specify the correct permission configuration in our templates: https://github.com/contentful/cf-frontend-app-template/blob/26ba419fc5b33f864fa115f7555735c1093d3202/.github/workflows/auto-approve-and-merge.yml.cd.hbs#L8-L10

In conclusion, I think it's as simple as remove the override and everything should "just work". Happy to try it out in my repo to verify it works correctly, so we can revert if things fail in the real world.

**Comment by jamesmbourne** [comment]
Yes you're totally right @jsdalton  - it's the responsiblity of the "host" action to specify permissions.

I think you can give the action a try as-is without needing to merge it - you should be able to reference a commit on this branch to test it out in another repo

**Comment by jsdalton** [comment]
Okay, I pushed that fix and tested it against the latest commit. This worked successfully (the bot merged this PR in my test: https://github.com/contentful/marketplace-partner-apps/pull/572 )

**Comment by jsdalton** [comment]
@jamesmbourne i believe this PR is ready for a ✅ if this all makes sense to you.

**Comment by jamesmbourne** [comment]
@jsdalton what about approval? Can we test that the action can also approve the PR?

**Comment by jsdalton** [comment]
@jamesmbourne checking now. (it's hard to convince our dependabot to open a fresh PR for approval 🤔 )

**Comment by jamesmbourne** [comment]
Great, thanks very much for checking it @jsdalton 🎉

**Review by jamesmbourne — APPROVED** [review]

**Comment by orYoffe** [comment]
FYI @jsdalton this broke some workflows so reverted here for the time being... wonder if we can just pass both for you or have another version 
https://github.com/contentful/github-auto-merge/pull/21
